### ISSUE TITLE
Focal Loss for Dense Object Detection

### ISSUE THREAD

**Issue by sile** [sile/readings#227]
URL
-----

https://arxiv.org/abs/1708.02002

Abstract
----------

> The highest accuracy object detectors to date are based on a two-stage approach popularized by R-CNN, where a classifier is applied to a sparse set of candidate object locations. In contrast, one-stage detectors that are applied over a regular, dense sampling of possible object locations have the potential to be faster and simpler, but have trailed the accuracy of two-stage detectors thus far. In this paper, we investigate why this is the case. We discover that the extreme foreground-background class imbalance encountered during training of dense detectors is the central cause. We propose to address this class imbalance by reshaping the standard cross entropy loss such that it down-weights the loss assigned to well-classified examples. Our novel Focal Loss focuses training on a sparse set of hard examples and prevents the vast number of easy negatives from overwhelming the detector during training. To evaluate the effectiveness of our loss, we design and train a simple dense detector we call RetinaNet. Our results show that when trained with the focal loss, RetinaNet is able to match the speed of previous one-stage detectors while surpassing the accuracy of all existing state-of-the-art two-stage detectors.

Table of Contents
--------------------

- (1) Introduction
- (2) Related Work
- (3) Focal Loss
  - (3-1) Balanced Cross Entropy
  - (3-2) Focal Loss Definition
  - (3-3) Class Imbalance and Model Initialization
  - (3-4) Class Imbalance and Two-stage Detectors
- (4) RetinaNet Detector
  - (4-1) Inference and Training
- (5) Experiments
  - (5-1) Training Dense Detection
  - (5-2) Model Architecture Design
  - (5-3) Comparison to State of the Art
- (6) Conclusion
- (A) Focal Loss*
- (B) Derivatives

BibTeX
------

```bibtex
@misc{lin2018focal,
      title={Focal Loss for Dense Object Detection}, 
      author={Tsung-Yi Lin and Priya Goyal and Ross Girshick and Kaiming He and Piotr Dollár},
      year={2018},
      eprint={1708.02002},
      archivePrefix={arXiv},
      primaryClass={cs.CV}
}
```

See Also
----------

- GitHub: https://github.com/facebookresearch/Detectron
